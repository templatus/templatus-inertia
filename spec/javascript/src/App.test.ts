import { mount } from '@vue/test-utils';

import App from '@/App.vue';

import { createRouter, createWebHistory } from 'vue-router';

const router = createRouter({
  history: createWebHistory(),
  routes: [
    {
      path: '/',
      component: { template: 'Home page' },
    },
    {
      path: '/about',
      component: { template: 'About page' },
    },
  ],
});

describe('App', () => {
  const wrapper = mount(App, {
    global: {
      plugins: [router],
    },
  });

  afterAll(() => {
    wrapper.unmount();
  });

  test('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('recognizes online/offline', async () => {
    window.dispatchEvent(new window.Event('offline'));
    await wrapper.vm.$nextTick();
    expect(wrapper.html()).toContain('offline');

    window.dispatchEvent(new window.Event('online'));
    await wrapper.vm.$nextTick();
    expect(wrapper.html()).not.toContain('offline');
  });
});
