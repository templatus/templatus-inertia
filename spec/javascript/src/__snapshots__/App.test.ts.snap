// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div class="flex min-h-screen flex-col bg-gradient-to-br from-rails-dark to-rails-light lg:bg-rose lg:from-inherit lg:to-inherit"><svg class="fixed inset-0 hidden h-96 w-full drop-shadow-xl lg:block" fill="none" preserveAspectRatio="none" viewBox="0 0 100 100">
    <defs>
      <linearGradient id="grad1" x1="0" y1="0" x2="100%" y2="0">
        <stop offset="0%" class="gradient__rails-dark"></stop>
        <stop offset="30%" class="gradient__rails-light"></stop>
        <stop offset="100%" class="gradient__rails-dark"></stop>
      </linearGradient>
    </defs>
    <polygon fill="url(#grad1)" points="0,0 0,100 100,40 100,0"></polygon>
  </svg><svg class="fixed bottom-0 hidden h-96 w-full drop-shadow-xl lg:block" fill="none" preserveAspectRatio="none" viewBox="0 0 100 100">
    <defs>
      <linearGradient id="grad2" x1="0" y1="0" x2="100%" y2="0">
        <stop offset="0%" class="gradient__rails-dark"></stop>
        <stop offset="60%" class="gradient__rails-light"></stop>
        <stop offset="100%" class="gradient__rails-dark"></stop>
      </linearGradient>
    </defs>
    <polygon fill="url(#grad2)" points="0,80 0,100 100,100 100,50"></polygon>
  </svg>
  <header class="flex items-end justify-between space-x-5 p-5 lg:flex-col lg:items-stretch lg:space-y-36 lg:space-x-0 lg:p-0 lg:fixed lg:top-10 lg:left-10"><img src="[object Object]" alt="Logo" class="h-20 w-20 rounded bg-gradient-to-br from-slate-200 to-white p-3 lg:h-56 lg:w-56" width="150" height="150">
    <!---->
    <nav role="tablist" aria-orientation="horizontal" class="flex space-x-5 text-center text-lg font-medium uppercase lg:flex-col lg:space-y-5 lg:space-x-0"><a href="/" class="router-link-active router-link-exact-active bg-white text-rails-dark lg:bg-rails-dark lg:text-white rounded-md px-2 py-1 lg:shadow" id="headlessui-tabs-tab-1" role="tab" aria-selected="true" tabindex="0" aria-current="page">Home</a><a href="/about" class="bg-transparent text-white hover:bg-slate-300 hover:text-rails-dark lg:bg-white lg:text-rails-dark rounded-md px-2 py-1 lg:shadow" id="headlessui-tabs-tab-2" role="tab" aria-selected="false" tabindex="-1">About</a></nav>
  </header>
  <main class="relative flex-1 bg-white py-5 px-4 shadow-md sm:px-6 lg:mt-10 lg:mr-16 lg:ml-96 lg:max-w-5xl lg:rounded-xl lg:px-8">Home page</main>
  <footer class="flex flex-wrap items-end justify-between space-y-2 px-4 text-sm text-white sm:px-6 lg:px-0 relative mt-2 mb-6 lg:mr-16 lg:ml-96 lg:max-w-5xl lg:px-0">
    <div class="flex w-full items-end space-x-2 sm:w-auto"><a class="hover:underline" target="_blank" rel="noopener noreferrer" href="https://github.com/ledermann/templatus"><svg viewBox="0 0 24 24" class="h-6 w-6" fill="currentColor">
          <title>GitHub</title>
          <path d="M12 .297c-6.63 0-12 5.373-12 12 0 5.303 3.438 9.8 8.205 11.385.6.113.82-.258.82-.577 0-.285-.01-1.04-.015-2.04-3.338.724-4.042-1.61-4.042-1.61C4.422 18.07 3.633 17.7 3.633 17.7c-1.087-.744.084-.729.084-.729 1.205.084 1.838 1.236 1.838 1.236 1.07 1.835 2.809 1.305 3.495.998.108-.776.417-1.305.76-1.605-2.665-.3-5.466-1.332-5.466-5.93 0-1.31.465-2.38 1.235-3.22-.135-.303-.54-1.523.105-3.176 0 0 1.005-.322 3.3 1.23.96-.267 1.98-.399 3-.405 1.02.006 2.04.138 3 .405 2.28-1.552 3.285-1.23 3.285-1.23.645 1.653.24 2.873.12 3.176.765.84 1.23 1.91 1.23 3.22 0 4.61-2.805 5.625-5.475 5.92.42.36.81 1.096.81 2.22 0 1.606-.015 2.896-.015 3.286 0 .315.21.69.825.57C20.565 22.092 24 17.592 24 12.297c0-6.627-5.373-12-12-12"></path>
        </svg></a>
      <!--v-if-->
    </div>
    <div class="w-full sm:w-auto">
      <!--v-if-->
    </div>
    <div class="flex w-full items-end sm:w-auto"><a class="hover:underline" target="_blank" rel="noopener noreferrer" href="https://ledermann.dev"><svg class="inline h-6 w-6" viewBox="0 0 276 276" fill="currentColor">
          <title>ledermann.dev</title>
          <path d="M0 137l138 138 138-138-84-82-84 82 31 30 53-52 25 22-79 77-78-77L166 29 138 1"></path>
        </svg><span class="ml-1 font-mono">ledermann.dev</span></a></div>
  </footer>
</div>
`;
